ui/src/alerting/components/notifications/StatusLevels.tsx

```diff
@@ -0,0 +1,71 @@
+// Libraries
+import React, {FC, useContext} from 'react'
+
+// Components
+import {
+  ComponentSpacer,
+  TextBlock,
+  FlexDirection,
+  ComponentSize,
+} from '@influxdata/clockface'
+import LevelsDropdown from 'src/alerting/components/notifications/LevelsDropdown'
+import StatusChangeDropdown from 'src/alerting/components/notifications/StatusChangeDropdown'
+import {NewRuleDispatch} from './NewRuleOverlay'
+
+// Types
+import {StatusRuleItem} from 'src/types'
+
+interface Props {
+  status: StatusRuleItem
+}
+
+const StatusLevels: FC<Props> = ({status}) => {
+  const {currentLevel, previousLevel} = status.value
+  const dispatch = useContext(NewRuleDispatch)
+
+  const onClickLevel = (type, level) => {
+    const value = {
```

Comment:
What do you think about moving this logic to the reducer with its own action type?

I think in general when following the action/reducer style, we are too reluctant to add actions for specific/one-off events that a user performs in the UI. The result is that we have very general `SET_FOO` actions that we dispatch in various configurations, and our reducers stay really small/dumb.  Then we don't get one of the big benefits of reducers: that they centralize state changes, so that we are coding an explicit rather than implicit state machine.

Not a big deal here, just something I wanna see us do differently in general.

Related thought-leadering: https://twitter.com/dan_abramov/status/800310164792414208

Reply:
That's what the ui used to do.  I don't now how we strayed.  But, I really like it.  Makes it super easy to test more complex state changes.